modules/tinymce/src/themes/silver/main/ts/ui/menus/item/build/CardMenuItem.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { AlloyComponent, AlloySpec, Behaviour, Disabling } from '@ephox/alloy';
+import { Menu } from '@ephox/bridge';
+import { Arr, Obj, Optional } from '@ephox/katamari';
+import { UiFactoryBackstageShared } from 'tinymce/themes/silver/backstage/Backstage';
+import { renderItemDomStructure } from 'tinymce/themes/silver/ui/menus/item/structure/ItemStructure';
+import ItemResponse from '../ItemResponse';
+import { renderContainer, renderDescription, renderHtml, renderImage } from '../structure/ItemSlices';
+import { replaceText } from './AutocompleteMenuItem';
+import { buildData, renderCommonItem } from './CommonMenuItem';
+
+interface CardExtras {
+  autocompleteMatchText?: string;
+  itemBehaviours?: Array<Behaviour.NamedConfiguredBehaviour<Behaviour.BehaviourConfigSpec, Behaviour.BehaviourConfigDetail>>;
+}
+
+const render = (items: Menu.CardItem[], extras: CardExtras): Array<AlloySpec> => Arr.map(items, (item) => {
+  switch (item.type) {
+    case 'container':
+      return renderContainer(render(item.items, extras), item.direction);
+
+    case 'image':
+      return renderImage(item.src, item.width, item.height, item.alt);
+
+    case 'description':
+      return renderDescription(item.text);
+
+    case 'autocomplete':
+      const matchText = Obj.get(extras, 'autocompleteMatchText').getOr('');
+      return renderHtml(replaceText(item.text, matchText));
+  }
+});
+
+export function renderCardMenuItem(
+  spec: Menu.CardMenuItem,
+  itemResponse: ItemResponse,
+  sharedBackstage: UiFactoryBackstageShared,
+  extras: CardExtras
+) {
+  const getApi = (component: AlloyComponent): Menu.CardMenuItemInstanceApi => ({
+    isDisabled: () => Disabling.isDisabled(component),
```

Comment:
This will only disable the "card" component, not the child components (eg title, description) I believe. Will that cause issues?

Reply:
There's no child component that got any interactivity so do they need to be disabled? Not sure we would have any either since it would be inside an autocompleter/menu item?

Reply:
While that's true atm, I'm concerned it'd easily get missed if we do add something.